torch/CMakeLists.txt

```diff
@@ -643,38 +180,37 @@ if (BUILD_PYTHON)
     # we're not careful.
     list(INSERT 0 TORCH_PYTHON_LINK_LIBRARIES ${CUDNN_LIBRARY})
     list(INSERT 0 TORCH_PYTHON_INCLUDE_DIRECTORIES ${CUDNN_INCLUDE_DIR})
-  endif()
+endif()
 
-  if (USE_MIOPEN)
+if (USE_MIOPEN)
     list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_MIOPEN)
     list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES ${MIOPEN_INCLUDE_DIR})
     list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${MIOPEN_LIBRARY})
-  endif()
+endif()
 
-  if (USE_NUMPY)
+if (USE_NUMPY)
     list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_NUMPY)
-  endif()
+endif()
 
-  if (USE_ROCM)
+if (USE_ROCM)
     list(APPEND TORCH_PYTHON_SRCS
       ${TORCH_SRC_DIR}/csrc/cuda/Module.cpp
       ${TORCH_SRC_DIR}/csrc/cuda/Storage.cpp
       ${TORCH_SRC_DIR}/csrc/cuda/Stream.cpp
       ${TORCH_SRC_DIR}/csrc/cuda/Event.cpp
       ${TORCH_SRC_DIR}/csrc/cuda/utils.cpp
-      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
```

Comment:
Is this is the reason causing the undefined symbol in rocm build? (although I don't understand why cuda build doesn't see the same issue).

Reply:
Ok I saw it has been moved to caffe2/CMakeLists.txt. But from the generated build.ninja file, comm.cpp is not included in the rocm build.